Add option to enable per endpoint routes

tgraf opened the pull request:
See individual commits for description

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/8089)
<!-- Reviewable:end -->

aanm commented on daemon/endpoint.go:
```diff
@@ -159,6 +159,15 @@ func (d *Daemon) errorDuringCreation(ep *endpoint.Endpoint, err error) (*endpoin
+		}
+
+		epTemplate.DatapathConfiguration.InstallEndpointRoute = true
+		epTemplate.DatapathConfiguration.RequireEgressProg = true
```
Add some comments here explaining that since there's no `cilium_host` to route the packets we need to install an egress program in host-facing egress.

aanm reviewed (changes requested):
Only one nit all good from the rest.

tgraf committed:
agent: Add --enable-endpoint-routes option

Add a flag --enable-endpoint-routes which enables use of per endpoint routes
for all new endpoints. The existing cilium_host interface is still created and
required for L7 proxy redirection at ingress. Ingress policy of a local
endpoint continues to be evaluated at egress of the sending pod unless
`--force-local-policy-eval-at-source=false` is passed to the agent.

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
test: Add CI test for --enable-endpoint-routes mode

Signed-off-by: Thomas Graf <thomas@cilium.io>
